NuGet.Config

```diff
@@ -54,12 +59,102 @@
       <package pattern="envdte80" />
       <package pattern="envdte90" />
       <package pattern="envdte90a" />
+      <package pattern="FluentAssertions" />
+      <package pattern="ILMerge" />
+      <package pattern="Lucene.Net" />
+      <package pattern="MessagePack" />
+      <package pattern="Microsoft.AspNetCore.App.Ref" />
+      <package pattern="Microsoft.Build" />
+      <package pattern="Microsoft.Build.*" />
+      <package pattern="Microsoft.Build.Artifacts" />
+      <package pattern="Microsoft.Build.Framework" />
+      <package pattern="Microsoft.Build.Locator" />
+      <package pattern="Microsoft.Build.Runtime" />
+      <package pattern="Microsoft.Build.Tasks.Core" />
+      <package pattern="Microsoft.Build.Utilities.Core" />
+      <package pattern="Microsoft.CodeAnalysis" />
+      <package pattern="Microsoft.CodeAnalysis.Analyzers" />
+      <package pattern="Microsoft.CodeAnalysis.AnalyzerUtilities" />
+      <package pattern="Microsoft.CodeAnalysis.BannedApiAnalyzers" />
+      <package pattern="Microsoft.CodeAnalysis.Common" />
+      <package pattern="Microsoft.CodeAnalysis.CSharp" />
+      <package pattern="Microsoft.CodeAnalysis.CSharp.Workspaces" />
+      <package pattern="Microsoft.CodeAnalysis.EditorFeatures.Text" />
+      <package pattern="Microsoft.CodeAnalysis.Features" />
+      <package pattern="Microsoft.CodeAnalysis.PublicApiAnalyzers" />
+      <package pattern="Microsoft.CodeAnalysis.Scripting.Common" />
+      <package pattern="Microsoft.CodeAnalysis.VisualBasic.Workspaces" />
+      <package pattern="Microsoft.CodeAnalysis.Workspaces.Common" />
+      <package pattern="Microsoft.CSharp" />
+      <package pattern="Microsoft.DataAI.NuGetRecommender.Contracts" />
+      <package pattern="Microsoft.DiaSymReader" />
+      <package pattern="Microsoft.DotNet.Build.Tasks.Feed" />
+      <package pattern="Microsoft.DotNet.Maestro.Tasks" />
+      <package pattern="Microsoft.Extensions.FileProviders.Abstractions" />
+      <package pattern="Microsoft.Extensions.FileSystemGlobbing" />
+      <package pattern="Microsoft.Extensions.Primitives" />
```

Comment:
DiaSymReader is a package published from the dotnet/runtime repo, so I can't imagine it needs to come from the vside feed (VS doesn't have a special version).

The Microsoft.Dotnet packages sound like dnceng packages that probably should be restored from dotnet-eng. In fact, the Build.Tasks.Feed package id this PR also adds an explicit mapping to for the dotnet-eng feed in addition to here for the vside feed.

The Microsoft.Extensions.* packages also are dotnet/* repo packages, so seem infeasible that they have VS specific versions.